pkg/controller/plan/kubevirt.go

```diff
@@ -2686,21 +2690,38 @@ func (r *VirtualMachine) Conditions() (cnd *libcnd.Conditions) {
 	return
 }
 
-// Create an OwnerReference from a VM.
-func vmOwnerReference(vm *cnv.VirtualMachine) (ref meta.OwnerReference) {
+// Generic function to create OwnerReferences based on the input type.
+func setOwnerReferences[T any](obj T) []meta.OwnerReference {
```

Comment:
nitpick, `setOwnerReferences` sounds like something the set the owner ref, e.g. it will set the owner ref of the obj,
but in this case it creates an owner ref obj, can you think of a name that will not have this issue ?